src/main/java/org/broadinstitute/hellbender/tools/walkers/variantutils/ValidateVariants.java

```diff
@@ -195,7 +200,14 @@ public void apply(final VariantContext vc, final ReadsContext readsContext, fina
         final Set<String> rsIDs = getRSIDs(featureContext);
 
         if (VALIDATE_GVCF) {
-            genomeLocSortedSet.add(genomeLocSortedSet.getGenomeLocParser().createGenomeLoc(ref.getInterval().getContig(), ref.getInterval().getStart(), vc.getEnd()), true);
+            final SimpleInterval refInterval = ref.getInterval();
+
+            // Take advantage of adjacent blocks and just merge them so we dont have to keep so many objects in the set.
+            final int start = (previousInterval != null && previousInterval.overlapsWithMargin(refInterval, 1)) ?
+                    previousInterval.getEnd() : refInterval.getStart();
```

Comment:
so if we notice the current interval is adjacent to the previous one, we "overlap" them before we add to the set